std-bits/table/src/main/java/org/enso/table/write/ExcelWriter.java

```diff
@@ -127,47 +126,99 @@ public static void writeTableToRange(Workbook workbook, ExcelRange range, boolea
       }
     }
 
-    if (range.isSingleCell()) {
-      ExcelRange expanded = ExcelRange.expandSingleCell(range, sheet);
+    ExcelRange expanded = range.isSingleCell() ? ExcelRange.expandSingleCell(range, sheet) : range;
+    headers = headers != ExcelHeaders.HeaderBehavior.INFER ? headers :
+        shouldWriteHeaders(sheet, expanded.getTopRow(), expanded.getLeftColumn(), expanded.getRightColumn());
 
-      if (headers == ExcelHeaders.HeaderBehavior.INFER) {
-        headers = shouldWriteHeaders(sheet, expanded.getTopRow(), expanded.getLeftColumn(), expanded.getRightColumn())
-            ? ExcelHeaders.HeaderBehavior.USE_FIRST_ROW_AS_HEADERS
-            : ExcelHeaders.HeaderBehavior.EXCEL_COLUMN_NAMES;
-      }
+    if ((existingDataMode == ExistingDataMode.APPEND_BY_NAME || existingDataMode == ExistingDataMode.APPEND_BY_INDEX) &&
+        rangeIsNotEmpty(workbook, expanded, false, sheet)) {
+      appendRangeWithTable(workbook, range, existingDataMode, table, rowLimit, headers, sheet, expanded);
+    } else {
+      updateRangeWithTable(workbook, expanded, range.isSingleCell(), existingDataMode, table, rowLimit, headers, sheet);
+    }
+  }
+
+  /**
+   * Creates an empty workbook.
+   * @param xls_format specifies whether the file is in Excel Binary Format (95-2003 format).
+   * @return a {@link Workbook} containing the specified data.
+   */
+  public static Workbook createWorkbook(boolean xls_format) {
+    return xls_format ? new HSSFWorkbook() : new XSSFWorkbook();
+  }
 
-      // Expand to cover required size
-      int rowCount = (headers == ExcelHeaders.HeaderBehavior.USE_FIRST_ROW_AS_HEADERS ? 1 : 0) + table.rowCount();
-      if (expanded.getColumnCount() < table.getColumns().length || expanded.getRowCount() < rowCount) {
-        expanded = new ExcelRange(
-            expanded.getSheetName(),
-            expanded.getLeftColumn(),
-            expanded.getTopRow(),
-            Math.max(expanded.getRightColumn(), expanded.getLeftColumn() + table.getColumns().length - 1),
-            Math.max(expanded.getBottomRow(), expanded.getTopRow() + rowCount - 1));
+  private static void appendRangeWithTable(Workbook workbook, ExcelRange range, ExistingDataMode existingDataMode, Table table, Long rowLimit, ExcelHeaders.HeaderBehavior headers, ExcelSheet sheet, ExcelRange expanded)
+      throws RangeExceededException, ExistingDataException, ColumnNameMismatchException, ColumnCountMismatchException {
+    // Map Table
```

Comment:
I don't think this comment is adding any meaningful information. It could be useful if it clarified what kind of mapping is performed, like `Reorder Table columns to match the found order in the file.`